src/woo_search/search_index/tasks.py

```diff
@@ -58,7 +58,7 @@ def index_document(
     download_url: str = "",
     file_size: int | None = None,
 ):
-    document = Document(
+    document = Document(  # pyright: ignore[reportCallIssue]
```

Comment:
Annoying side-effect from setting up the attachment field the way we do. Maybe there is a different way to do so?

Reply:
I don't think that type definition is correct - `InnerDoc` is not something you'd pass directly to the `Document(...)` instantiation.

Reply:
with my above experiments, we can drop the `attachment` type definition and this workaround will then also not be necessary anymore